tests/integration/api/mat_pulsar_to_pulsar_test.go

```diff
@@ -200,11 +200,21 @@ var _ = PDescribe("Materialization from Pulsar to Pulsar", Ordered, Label("pulsa
 				var data pulsarToPulsarTestSchema
 				err = json.Unmarshal(msg.Payload(), &data)
 				g.Expect(err).ShouldNot(HaveOccurred())
-				g.Expect(data.LastId).Should(Equal(1))
-				g.Expect(data.LastTime).Should(Equal(20))
-				g.Expect(data.Count).Should(Equal(2))
-
+				g.Expect(data.LastId).Should(Equal(2))
+				g.Expect(data.LastTime).Should(Equal(1687303803000000000))
+				g.Expect(data.Count).Should(Equal(1))
 				g.Expect(pulsarConsumer.Ack(msg)).Should(Succeed())
+
+				msg2 := receivePulsarMessageWithTimeout(pulsarConsumer, ctx)
+				g.Expect(msg2).ShouldNot(BeNil())
+
+				var data2 pulsarToPulsarTestSchema
+				err = json.Unmarshal(msg2.Payload(), &data2)
+				g.Expect(err).ShouldNot(HaveOccurred())
+				g.Expect(data2.LastId).Should(Equal(4))
+				g.Expect(data2.LastTime).Should(Equal(1687303805000000000))
+				g.Expect(data2.Count).Should(Equal(1))
```

Comment:
Hm. We're getting a second message? I would've expected not, since we shouldn't be processing the second input row with the current late data handling logic. I wonder if this is because of our `send_batch` addition? But if the values here are correct, we are getting new results (although..still 1 count?)

Reply:
I believe expecting the second message is correct. When we start the test, we load two messages and the first is read at "should output initial results to pulsar". The second event is in the buffer. Then we load two more (3 + 4) events in the "load more data into the table" test. Since each event is over 1 second apart, the events 2 and 3 should be processed while 4 is left in the buffer.

Reply:
Ah, you're correct - didn't realize this was under the `load additional data` part